### ISSUE TITLE
How to configure a specific load balancer for DTR

### ISSUE THREAD

**Issue by joaofnfernandes** [docker/docker.github.io#1703]
We have an [overview page ](https://docs.docker.com/datacenter/dtr/2.2/guides/admin/configure/use-a-load-balancer/), but we need docs with example configurations for specific load balancers.

- [ ] haproxy
- [ ] elb
- [ ] nginx

Before starting, create a new issue where you specify the load balancer config you're going to tackle, and reference this issue on the new one.

**Comment by pdevine** [comment]
This should include how the /health check endpoint works.  We need to be explicit about how to get per replica health, as well as cluster health.

**Comment by nicolaka** [comment]
dibs

**Comment by joaofnfernandes** [comment]
@nicolaka if you're tackling just a single load balancer, create a separate issue to track that. But given your chops I bet you'll try to have config files for all three :D

**Comment by nicolaka** [comment]
anyone has an idea why in haproxy https healthchecks are failing with the following config?

```
backend dtr_upstream_servers_443
        mode tcp
        option httpchk /health
        server node01:12391 172.31.39.165:12391 check check-ssl verify none weight 100
        server node02:12391 172.31.21.168:12391 check check-ssl verify none weight 100
        server node03:12391 172.31.14.53:12391 check check-ssl verify none weight 100
```

@pdevine @vikstrous ?

**Comment by whoshuu** [comment]
By default `option httpchk` uses `OPTIONS` as the HTTP method. `OPTIONS` is not a valid method on that endpoint. Try this instead:

```
option httpchk get /health
```

**Comment by nicolaka** [comment]
same result @whoshuu with :

```
backend dtr_upstream_servers_443
        mode tcp
        option httpchk get /health
        server node01 172.31.39.165:12391 weight 100 check
        server node02 172.31.21.168:12391 weight 100 check
        server node03 172.31.14.53:12391 weight 100 check
```

if i check directly with curl it also doesn't work unless I add `https://` to the URL.

**Comment by vikstrous** [comment]
This is the template we use:

```
frontend localnodes
	bind *:%s
	mode tcp
	default_backend nodes
	timeout client 1h

backend nodes
	mode tcp
	balance roundrobin
	{{range .}}
	server clone{{.Index}} {{.IP}}:{{.Port}} check
	{{end}}
	timeout connect 20s
	timeout server 1h
```

**Comment by nicolaka** [comment]
Still no luck here. I think this should be the right setting to USE https to do healthchecks but without verifying server cert.

```
backend dtr_upstream_servers_443
        mode tcp
        option httpchk GET /health HTTP/1.0
        server node01 172.31.39.165:12391 weight 100 check ssl verify none
```

Anyone can help?

**Comment by nicolaka** [comment]
@pdevine @vikstrous this exact config is working fine with UCP's `/_ping`, what could be different with `/health` ? is it ciphers/ssl/tls versions required?

**Comment by nicolaka** [comment]
@mbentley pointed out need to add Host info. 

```
option httpchk GET /health HTTP/1.1\r\nHost:\ <DTR_FQDN>
```

So we should be good.

**Comment by joaofnfernandes** [comment]
@nicolaka do you have a PR for this?

**Comment by nicolaka** [comment]
I'm working on it now should have it by end of day @joaofnfernandes .
